src/hostname/hostnamed.c

```diff
@@ -250,8 +250,30 @@ static int get_firmware_vendor(char **ret) {
          return get_hardware_firmware_data("bios_vendor", ret);
 }
 
-static int get_firmware_date(char **ret) {
-         return get_hardware_firmware_data("bios_date", ret);
+static int get_firmware_date(usec_t *ret) {
+         _cleanup_free_ char *bios_date = NULL;
+         int r;
+
+         r = get_hardware_firmware_data("bios_date", &bios_date);
+         if (r <= 0)
+                 return r;
+         if (r == 0) {
+                 *ret = USEC_INFINITY;
+                 return 0;
+         }
+
+         for (size_t i = 0; i <= strlen(bios_date); i++) {
+                 if (bios_date[i] == '/')
+                         bios_date[i] = '-';
+         }
+
+        r = parse_timestamp(bios_date, ret);
```

Comment:
please don't use parse_timestamp(), we should be strict here, and only accept a strict format, with the expected syntax.

in particular as parse_timestamp expects the format `YYYY-MM-DD` while the sysfs property outputs `MM/DD/YYYY` no matter of converting `/` to `-` is going to rearrange the numbers for you...

Reply:
Applied the parsing suggestion, still an issue left with `hostnamectl` although the timestamp returned via DBUS looks correct.

Reply:
"an issue"?